internal/cephfs/volume.go

```diff
@@ -260,3 +277,27 @@ func purgeVolume(ctx context.Context, volID volumeID, cr *util.Credentials, volO
 
 	return nil
 }
+
+// checkSubvolumeHasFeature verifies if the referred subvolume has
+// the required feature.
+func checkSubvolumeHasFeature(ctx context.Context, feature string, volID volumeID, cr *util.Credentials, volOptions *volumeOptions) (bool, error) {
+	info, err := getSubVolumeInfo(ctx, volOptions, cr, volID)
+	if err != nil {
+		// For older Ceph versions where 'features' was not returned in the info output,
+		// or `subvolume info` command is not supported,it is assumed that these do not
+		// support the intended features.
+		// The Purge operation should then continue without snapshot retention.
+		if errors.Is(err, ErrInvalidCommand) {
+			return false, nil
+		}
+		return false, status.Error(codes.Internal, err.Error())
```

Comment:
we should not return the GRPC error code from volume.go file

Reply:
Oh my bad. Will get rid of it.